includes/class-wc-payments-account.php

```diff
@@ -842,17 +882,115 @@ public function maybe_redirect_from_onboarding_page(): bool {
+	 *       Connect links are used to start/re-start/continue the onboarding flow and they are independent of
+	 *       the WP dashboard page (based solely on request params).
+	 *
+	 * IMPORTANT: The logic should be kept in sync with the one in maybe_redirect_from_overview_page to avoid loops.
+	 *
+	 * @see self::maybe_redirect_from_overview_page() for the opposite redirection.
+	 * @see self::maybe_handle_onboarding() for connect links handling.
+	 *
+	 * @return bool True if the redirection happened, false otherwise.
+	 */
+	public function maybe_redirect_from_connect_page(): bool {
+		if ( wp_doing_ajax() || ! current_user_can( 'manage_woocommerce' ) ) {
+			return false;
+		}
+
+		$params = [
+			'page' => 'wc-admin',
+			'path' => '/payments/connect',
+		];
+
+		// We're not on the Connect page, don't redirect.
+		if ( count( $params ) !== count( array_intersect_assoc( $_GET, $params ) ) ) { // phpcs:disable WordPress.Security.NonceVerification.Recommended
+			return false;
+		}
+
+		// If everything is in good working condition, redirect to Payments Overview page.
+		if ( $this->has_working_jetpack_connection() && $this->is_stripe_account_valid() ) {
+			$this->redirect_service->redirect_to_overview_page( WC_Payments_Onboarding_Service::FROM_CONNECT_PAGE );
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Redirects overview page (payments/overview) to the connect page for stores that
+	 * don't have a working Jetpack connection or a valid connected Stripe account.
+	 *
+	 * IMPORTANT: The logic should be kept in sync with the one in maybe_redirect_from_connect_page to avoid loops.
+	 *
+	 * @see self::maybe_redirect_from_connect_page() for the opposite redirection.
+	 * @see self::maybe_handle_onboarding() for connect links handling.
+	 *
+	 * @return bool True if the redirection happened, false otherwise.
+	 */
+	public function maybe_redirect_from_overview_page(): bool {
+		if ( wp_doing_ajax() || ! current_user_can( 'manage_woocommerce' ) ) {
+			return false;
+		}
+
+		$params = [
+			'page' => 'wc-admin',
+			'path' => '/payments/overview',
+		];
+
+		// We're not on the Overview page, don't redirect.
+		if ( count( $params ) !== count( array_intersect_assoc( $_GET, $params ) ) ) { // phpcs:disable WordPress.Security.NonceVerification.Recommended
+			return false;
+		}
+
+		// If everything is NOT in good working condition, redirect to Payments Connect page.
+		if ( ! $this->has_working_jetpack_connection() || ! $this->is_stripe_account_valid() ) {
+			$this->redirect_service->redirect_to_connect_page(
+				sprintf(
+				/* translators: 1: WooPayments. */
+					__( 'Please <b>complete your %1$s setup</b> to process transactions.', 'woocommerce-payments' ),
+					'WooPayments'
+				),
+				WC_Payments_Onboarding_Service::FROM_OVERVIEW_PAGE,
+				[
+					'promo'                       => ! empty( $_GET['promo'] ) ? sanitize_text_field( wp_unslash( $_GET['promo'] ) ) : false,
```

Comment:
I wonder if we need `promo`, `progressive`, `collect_payout_requirements` and `create_builder_account` on the connect page

Reply:
You have a point! Probably not.